jlibra-core/src/main/java/dev/jlibra/admissioncontrol/query/GetAccountTransactionBySequenceNumber.java

```diff
@@ -2,11 +2,28 @@
 
 import org.immutables.value.Value;
 
+import com.google.protobuf.ByteString;
+
+import types.GetWithProof.GetAccountTransactionBySequenceNumberRequest;
+import types.GetWithProof.RequestItem;
+
 @Value.Immutable
 public interface GetAccountTransactionBySequenceNumber {
 
     byte[] getAccountAddress();
 
     long getSequenceNumber();
 
+    default RequestItem toGrpcObject() {
```

Comment:
Honestly, all these `default` methods implementing serialization look smelly to me. It works and it is the best solution for the _Immutable_ pattern, but to me it's not like a default method should be used. There are many situations this pattern would break, eg. I would implement the interface and just forget about the default method. The serialization would then most likely be wrong and I wouldn't even recognize it.  
I understand moving the serialization to the domain objects, but this solution shouldn't be the final one IMHO.

Reply:
It's true that we probably would not have these implemented as interfaces if it wasn't for the Immutables library but still I don't see the problem with the default method. I think of it as something similar to abstract classes where you have unimplemented abstract methods and also concrete methods.
If we would actually implement the interface (instead of Immutables doing it), we would implement those two methods defined in the interface and the default method would still work fine, I don't see what could be the problem with forgetting about the default method?
Immutables supports also having these as abstract classes and I have used that for example in the `UpdateToLatestLedgerResult` class, that was because I also had to include private methods. I could consider converting these others also to abstract classes but I would first like to understand better the problem.

Reply:
To me, a "strategy" like serialization is difficult to set into the interface as a default method. For example, let's say interface A implements the default serialization. If I now implement it in class B, the default method will be the same for all implementations, but class B maybe adds 2 new fields and class C adds 3 other fields and still both use the "generic" serialization from the interface, which probably will break both implementations.  
I know this Immutables interface is not meant as a interface per se, but with this behaviour it's strange. With abstract classes this might be more visible, that seems less fishy to me...

Reply:
I changed these to abstract classes, you're right it's cleaner this way